nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -526,14 +622,20 @@ def __init__(
 
     def tie_weights(self,):
         pos_idx = 0
-        layer0 = self.model.language_model.encoder.layers[0]
+
+        if self.mcore_gpt:
+            layers = self.model.decoder.layers
+        else:
+            layers = self.model.language_model.encoder.layers
```

Comment:
Do we need to check if self.model is being wrapped here?

Reply:
done.